x/stakeibc/keeper/icqcallbacks_withdrawal_reward_balance.go

```diff
@@ -54,14 +58,34 @@ func WithdrawalRewardBalanceCallback(k Keeper, ctx sdk.Context, args []byte, que
 	k.Logger(ctx).Info(utils.LogICQCallbackWithHostZone(chainId, ICQCallbackID_WithdrawalRewardBalance,
 		"Query response - Withdrawal Reward Balance: %v %s", withdrawalRewardBalanceAmount, tradeRoute.RewardDenomOnHostZone))
 
-	// Using ICA commands on the withdrawal address, transfer the found reward tokens from the host zone to the trade zone
-	if err := k.TransferRewardTokensHostToTrade(ctx, withdrawalRewardBalanceAmount, tradeRoute); err != nil {
+	// Split the withdrawal amount into a rebate, stride fee, and reinvest portion
+	feeInfo, err := k.CheckForCommunityPoolRebate(ctx, chainId, withdrawalRewardBalanceAmount)
+	if err != nil {
+		return errorsmod.Wrapf(err, "unable to check for community pool rebate")
+	}
+
+	// If there's a rebate portion, fund the community pool with that amount
+	if feeInfo.RebateAmount.GT(sdkmath.ZeroInt()) {
+		rebateToken := sdk.NewCoin(tradeRouteCallback.RewardDenom, feeInfo.RebateAmount)
+		if err := k.FundCommunityPool(ctx, hostZone, rebateToken, types.ICAAccountType_WITHDRAWAL); err != nil {
+			return errorsmod.Wrapf(err, "unable to submit fund community pool ICA")
+		}
+
+		k.Logger(ctx).Info(utils.LogICQCallbackWithHostZone(chainId, ICQCallbackID_WithdrawalRewardBalance,
+			"Sending rebate tokens %v %s to community pool",
+			feeInfo.RebateAmount, tradeRoute.RewardDenomOnHostZone))
+	}
+
+	// Transfer the amount leftover after to the rebate to the trade zone so it can be swapped for the native token
+	// We transfer both the amount to be reinvested, and the amount for the stride fee
+	tradeAmount := feeInfo.ReinvestAmount.Add(feeInfo.StrideFeeAmount)
+	if err := k.TransferRewardTokensHostToTrade(ctx, tradeAmount, tradeRoute); err != nil {
 		return errorsmod.Wrapf(err, "initiating transfer of reward tokens to trade ICA failed")
 	}
 
 	k.Logger(ctx).Info(utils.LogICQCallbackWithHostZone(chainId, ICQCallbackID_WithdrawalRewardBalance,
 		"Sending discovered reward tokens %v %s from hostZone to tradeZone",
-		withdrawalRewardBalanceAmount, tradeRoute.RewardDenomOnHostZone))
+		feeInfo.ReinvestAmount, tradeRoute.RewardDenomOnHostZone))
```

Comment:
I think this should be `tradeAmount` instead of `feeInfo.ReinvestAmount`
and `tradeRoute.RewardDenomOnRewardZone` instead of `tradeRoute.RewardDenomOnHostZone` which is ibc

Reply:
done